rudder-core/src/main/scala/com/normation/rudder/services/policies/JavascriptEngine.scala

```diff
@@ -0,0 +1,604 @@
+    def bindings = toBindings("rudder", new JsRudderLibImpl(CryptHash))
+  }
+}
+
+
+/**
+ * This class provides the Rhino (java 7) or Longhorn (Java 8 & up)
+ * java script engine.
+ *
+ * It allows to eval parameters in directives which are starting
+ * with $eval.
+ *
+ */
+final object JsEngineProvider {
+
+
+  /**
+   * Initialize a new JsEngine with the correct bindings.
+   *
+   * Not all Libs (bindings in JSR-223 name) can't be provided here,
+   * because we want to make them specific to each eval
+   * (i.e: different eval may have to different bindings).
+   * So we just provide eval-indep lib here, but we don't
+   * have any for now.
+   */
+  def withNewEngine[T](feature: FeatureSwitch)(script: JsEngine => Box[T]): Box[T] = {
+    feature match {
+      case FeatureSwitch.Enabled  =>
+        SandboxedJsEngine.sandboxed { engine => script(engine) }
+      case FeatureSwitch.Disabled =>
+        script(DisabledEngine)
+    }
+  }
+
+}
+
+sealed trait JsEngine {
+  /**
+   *
+   * Parse a value looking for EVAL keyword.
+   * Return the correct NodeContextualizedValue type.
+   *
+   * Note that the eval result is ALWAY cast to
+   * string. So computation, object, etc must be
+   * correctly defined by the user to get and
+   * interesting value.
+   */
+  def eval(variable: Variable, lib: JsRudderLibBinding): Box[Variable]
+}
+
+/*
+ * Our JsEngine.
+ */
+final object JsEngine {
+  final val EVAL = "$evaljs "
+  final val EVAL_SIZE = EVAL.size
+
+  final object DisabledEngine extends JsEngine {
+    /*
+     * Eval does nothing on variable without the EVAL keyword, and
+     * fails on variable with the keyword.
+     */
+    def eval(variable: Variable, lib: JsRudderLibBinding): Box[Variable] = {
+      variable.values.find( _.startsWith(EVAL) ) match {
+        case None    => Full(variable)
+        case Some(v) =>
+          /*
+           * Here, we need to chose between:
+           * - fails when the feature is disabled, but the string starts with $eval,
+           *   meaning that maybe the user wanted to use it anyway.
+           *   But that means that we are changing GENERATION behavior on existing prod,
+           *   for a feature the user don't know anything.
+           * - not fails, because perhaps the user had that in its parameter. But in
+           *   that case, it will fails when feature is enabled by default.
+           *   And we risk to let the user spread sensitive information  into nodes
+           *   (because he thought the will be hashed, but in fact no).
+           *
+           * For now, failing because it seems to be the safe bet.
+           */
+          Failure(s"Value '${v}' starts with the ${EVAL} keyword, but the 'parameter evaluation feature' "
+                  +"is disable. Please, either don't use the keyword or enable the feature")
+      }
+    }
+  }
+
+  final object SandboxedJsEngine {
+     val MAX_EVAL_DURATION = (5, TimeUnit.SECONDS)
```

Comment:
The value is purelly arbitrary. We expects that a normal use case ends in tens of ms. But we don't want the user to have a whole generation fails because it scripts took 2 seconds for reasons. 
As it is something rather exceptionnal, and which will ends the Policy Generation, the value can be rather hight.